packages/flutter_tools/lib/src/codegen.dart

```diff
@@ -0,0 +1,291 @@
+        mainPath: mainPath,
+        targetProductVm: targetProductVm,
+        extraFrontEndOptions: extraFrontEndOptions
+      );
+      final File outputFile = fs.file(outputFilePath);
+      if (!await outputFile.exists()) {
+        await outputFile.create();
+      }
+      await outputFile.writeAsBytes(await buildResult.dillFile.readAsBytes());
+      return CompilerOutput(outputFilePath, 0);
+    } on Exception catch (err) {
+      printError('Compilation Failed: $err');
+      return const CompilerOutput(null, 1);
+    }
+  }
+}
+
+/// An implementation of a [ResidentCompiler] which runs a [BuildRunner] before
+/// talking to the frontend_server.
+class CodeGeneratingResidentCompiler implements ResidentCompiler {
+  CodeGeneratingResidentCompiler._(this._residentCompiler, this._codegenDaemon);
+
+  /// Creates a new [ResidentCompiler] and configures a [BuildDaemonClient] to
+  /// run builds.
+  static Future<CodeGeneratingResidentCompiler> create({
+    @required String mainPath,
+    bool trackWidgetCreation = false,
+    CompilerMessageConsumer compilerMessageConsumer = printError,
+    bool unsafePackageSerialization = false,
+  }) async {
+    final FlutterProject flutterProject = await FlutterProject.current();
+    final CodegenDaemon codegenDaemon = await codeGenerator.daemon(
+      extraFrontEndOptions: <String>[],
+      linkPlatformKernelIn: false,
+      mainPath: mainPath,
+      targetProductVm: false,
+      trackWidgetCreation: trackWidgetCreation,
+    );
+    codegenDaemon.startBuild();
+    await codegenDaemon.buildResults.firstWhere((bool result) => result);
+    final ResidentCompiler residentCompiler = ResidentCompiler(
+      artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
+      trackWidgetCreation: trackWidgetCreation,
+      packagesPath: codegenDaemon.packagesFile.path,
+      fileSystemRoots: <String>[
+        flutterProject.generated.absolute.path,
+        flutterProject.directory.path,
+      ],
+      fileSystemScheme: _kMultiRootScheme,
+      targetModel: TargetModel.flutter,
+      unsafePackageSerialization: unsafePackageSerialization,
+    );
+    return CodeGeneratingResidentCompiler._(residentCompiler, codegenDaemon);
+  }
+
+  final ResidentCompiler _residentCompiler;
+  final CodegenDaemon _codegenDaemon;
+
+  @override
+  void accept() {
+    _residentCompiler.accept();
+  }
+
+  @override
+  Future<CompilerOutput> compileExpression(String expression, List<String> definitions, List<String> typeDefinitions, String libraryUri, String klass, bool isStatic) {
+    return _residentCompiler.compileExpression(expression, definitions, typeDefinitions, libraryUri, klass, isStatic);
+  }
+
+  @override
+  Future<CompilerOutput> recompile(String mainPath, List<String> invalidatedFiles, {String outputPath, String packagesFilePath}) async {
+    _codegenDaemon.startBuild();
+    await _codegenDaemon.buildResults.first;
+    // This might be a bug
```

Comment:
I think the comment could be a bit more specific. I'm guessing a file should not already exist.